specs/EmergencyAuthCodeAPI.yml

```diff
@@ -0,0 +1,226 @@
+        '401':
+          description: Unauthorised
+        '404':
+          description: Not found
+  /emergency-auth-code-service/auth-code-requests:
+    post:
+      tags:
+        - auth-code-requests
+      operationId: createAuthCodeRequest
+      summary: Create an emergency auth code request
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/emergencyAuthCodeRequest'
+        description: Emergency auth code request data
+        required: true
+      responses:
+        '201':
+          description: Created emergency auth code request
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/emergencyAuthCodeRequest'
+        '400':
+          description: Bad request
+        '401':
+          description: Unauthorised
+  /emergency-auth-code-service/auth-code-requests/{auth_code_request_id}:
+    parameters:
+      - $ref: '#/components/parameters/authCodeRequestId'
+    get:
+      tags:
+        - auth-code-requests
+      operationId: getAuthCodeRequest
+      summary: Get an emergency auth code request
+      responses:
+        '200':
+          description: Updated emergency auth code request
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/emergencyAuthCodeRequest'
+        '400':
+          description: Bad request
+        '401':
+          description: Unauthorised
+    put:
+      tags:
+        - auth-code-requests
+      operationId: updateAuthCodeRequest
+      summary: Update an emergency auth code request
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/emergencyAuthCodeRequest'
+        description: Emergency auth code request data
+        required: true
+      responses:
+        '200':
+          description: Updated emergency auth code request
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/emergencyAuthCodeRequest'
+        '400':
+          description: Bad request
+        '401':
+          description: Unauthorised
+components:
+  schemas:
+    companyOfficer:
+      type: object
+      required:
+        - id
+        - forename
+        - surname
+      properties:
+        id:
+          type: string
+          description: The id of the user
+          readOnly: true
+          example: abc123
+        forename:
+          type: string
+          description: The officers forename
+          example: Joe
+        surname:
+          type: string
+          description: The officers surname
+          example: Bloggs
+    companyOfficers:
+      type: object
+      required:
+        - items
+        - total_count
+      properties:
+        items:
+          $ref: '#/components/schemas/companyOfficer'
```

Comment:
When returning multiple officers should this be an array?

Reply:
nice catch, updated to an array and tweaked the surround count field structure to use standard pagination fields